### DISCUSSION TITLE
Q-table cors issue on filter

### DISCUSSION THREAD

**Discussion by viniciusccarvalho** [quasarframework/quasar#11859] [General - Components / Directives / etc] [answered]
Hi there, I have an app deployed on firebase, and the behavior I'm currently seeing is something new, not sure if its due firebase function updates or quasar. Please indulge me the benefit of doubt:

my template looks like this:

```html
 <q-table
        :title="$t('patient.record.title')"
        :data="patients"
        :columns="columns"
        :pagination.sync="pagination"
        :filter="terms"
        @request="getPatients"
        row-key="_id"
        @row-click="selectRow"
        :loading="loading"
        binary-state-sort
      >
       <template v-slot:top-right>
          <q-input
            borderless
            dense
            debounce="300"
            clearable
            v-model="terms"
            :placeholder="$t('patient.table.filter')"
          >
            <template v-slot:append>
              <q-icon name="search"></q-icon>
            </template>
          </q-input>
        </template>
        <template v-slot:loading>
          <q-inner-loading
            showing
            color="primary"
          />
        </template>
      </q-table>
```
and the `getPatients` function:

```javascript
 getPatients (props) {

      const { page, rowsPerPage, sortBy, descending } = props.pagination
      const search = props.search
      const filterQuery = this.filter ? "INTERNO" : undefined
      service.listPatients({ terms: this.terms, page: page, limit: rowsPerPage, filter: filterQuery }).then(res => {
        this.pagination.rowsNumber = res.total
        this.pagination.page = res.page
        this.patients = res.items
        this.loading = false
      }).catch(err => {
        this.loading = false
        console.log("err", err)
        this.$q.notify({
          message: this.$t('errors.serverInternal'),
          color: 'negative'
        })

      })
    },
mounted () {
    this.getPatients({ pagination: this.pagination, search: this.terms })
  }
```

the `service` is a wrapper around the firebase functions:

```javascript
listPatients (query) {
        return new Promise((resolve, reject) => {
            let listFunction = this.functions.httpsCallable("listPatients")
            listFunction(query).then((result) => {
                resolve(result.data)
            }).catch(error => {
                console.log(error)
                reject(error)
            })
        })
    }
```

When the page loads, or when navigating the pages via the component next/previous it works just fine. But the moment I try to filter, there's an error from the server complaining about CORS:
```Access to fetch at 'https://*******.cloudfunctions.net/listPatients' from origin 'https://****.web.app' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.```

What is odd to me is that this was working a few months ago, the last time I checked, and the fact that only happens when there's a change on the filters data model.

Any ideas if this is something to be configured at quasar level? 

Cheers

**Comment by MilosPaunovic** [comment]
Hello, this is entirely `Firebase` issue, so you should focus your search on that.

**Comment by viniciusccarvalho** [comment]
I'm not entirely certain on that yet, because the first request and the paging always work. This is the bit that is quite odd to me, why would those request work, CORS would prevent them to work entirely. I'm wondering if the q-table component during the change in the `:filter` binding is missing something, or sending another request such as `OPTION` before doing a `GET`. I've seen errors on `OPTION` requests and CORS on `firebase`

**Comment by pdanpdan** [comment]
Cors is only triggered when some conditions are matched:
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests

**Comment by metalsadman** [comment]
this line looks to be the culprit `const filterQuery = this.filter ? "INTERNO" : undefined`, you should check on your network tab what params are sent thru the request when you are filtering.

**Reply by viniciusccarvalho** [reply]
It wasn't that, the filter is sent. It was a firebase issue, but on the function side, the database connection was throwing an error on a missed index, but instead of returning the error the browser for some reason was displaying a CORS error. This wasn't an http function, but a call function.

**Comment by viniciusccarvalho** [comment, marked as answer]
Problem was with database index, browser for some reason instead of showing a 500 error, was reporting CORS errors.
